model/patch_lifecycle.go

```diff
@@ -425,9 +426,156 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 	if err = p.SetActivated(patchVersion.Id); err != nil {
 		return nil, errors.WithStack(err)
 	}
+
+	if err = processTriggerAliases(ctx, p, projectRef); err != nil {
+		return nil, errors.Wrap(err, "problem processing trigger aliases")
+	}
+
 	return patchVersion, nil
 }
 
+func processTriggerAliases(ctx context.Context, p *patch.Patch, projectRef *ProjectRef) error {
+	if len(p.Triggers.Aliases) == 0 {
+		return nil
+	}
+
+	type aliasGroup struct {
+		project        string
+		status         string
+		parentAsModule string
+	}
+	aliasGroups := make(map[aliasGroup][]PatchTriggerDefinition)
+	for _, aliasName := range p.Triggers.Aliases {
+		aliasFound := false
+		var alias PatchTriggerDefinition
+		for _, alias = range projectRef.PatchTriggerAliases {
+			if alias.Alias == aliasName {
+				aliasFound = true
+				break
+			}
+		}
+		if !aliasFound {
+			// the user specified a non-existent alias
+			continue
```

Comment:
does it make sense to have this return an error? That way the CLI or web UI can display it

Reply:
done.